src/picterra/client.py

```diff
@@ -301,6 +306,58 @@ def list_rasters(self, folder_id=None):
         params = {'folder': folder_id} if folder_id else {}
         return self._paginate_through_list('rasters', params)
 
+    def edit_raster(
+        self, raster_id: str,
+        name: Optional[str]=None, folder_id: Optional[str]=None,
+        captured_at: Optional[str]=None, identity_key:Optional[str]=None,
+        multispectral_band_specification: Optional[dict] = None
+    ):
+        """
+        Edits an already existing raster.
+
+        Args:
+            name: New human-readable name for this raster
+            folder_id: Id of the new folder for this raster (move is in another project)
+            captured_at: new ISO-8601 date and time at which this
+                raster was captured, YYYY-MM-DDThh:mm[:ss[.uuuuuu]][+HH:MM|-HH:MM|Z];
+                e.g. "2020-01-01T12:34:56.789Z"
+            identity_key: New personal identifier for this raster.
+            multispectral_band_specification: If True, the raster is in multispectral mode and can have an associated band specification
+
+        Returns:
+            raster_id (str): The id of the uploaded raster
+        """
+        data = {}
+        if name is not None:
+            if len(name) == 0:
+                raise ValueError('Invalid empty name')
+            data.update({'name': name})
+        if folder_id is not None:
+            data.update({'folder_id': folder_id})
+        if captured_at is not None:
+            data.update({'captured_at': captured_at})
+        if identity_key is not None:
+            data.update({'identity_key': identity_key})
+        if multispectral_band_specification is not None:
+            if not isinstance(multispectral_band_specification, dict):
+                raise ValueError('Invalid multispectral band specification: not a dict')
+            keys = multispectral_band_specification.keys()
+            if not ('ranges' in keys and'vizbands' in keys):
+                raise ValueError('Invalid multispectral band specification: missing keys')
+            ranges = multispectral_band_specification['ranges']
+            vizbands = multispectral_band_specification['vizbands']
+            if not (isinstance(ranges, list) and isinstance(vizbands, list)):
+                raise ValueError('Invalid multispectral band specification: not arrays')
+            if len(vizbands) != 3:
```

Comment:
I feel we are doing too fine-grained validation client-side here. Those are things the server should return through a ValidationError, but doing this client-side means I feel we're going to have desync between client logic and server logic. So I wouldn't have bandspec validation here.

(But let's make sure we return a readable ValidationError from the backend when a bandspec is invalid)

Reply:
yes my bad, fixed (we actually check it in the be and raise a proper error)

Reply:
Thanks!